### PR TITLE
[BEAM-1795] Upgrade google-cloud-bigquery to 0.23.0

### PR DESCRIPTION
**Pull request by markflyhigh**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

Update google-cloud-bigquery to next stable version. No additional dependency is introduced after transitive dependency inspection.

### CODE REVIEW

File: sdks/python/setup.py [hunk 1]
```diff
@@ -106,7 +106,7 @@ def get_version():
   'proto-google-cloud-datastore-v1==0.90.0',
   'googledatastore==7.0.0',
   # GCP packages required by tests
-  'google-cloud-bigquery>=0.22.1,<0.23',
+  'google-cloud-bigquery>=0.23.0,<1.0.0',
```
Comment (aaltay) [inline sdks/python/setup.py]: Please put the upper bound to `0.24.0`. With major version `0` it is possible to introduce breaking changes with minor version changes.

Comment (markflyhigh) [inline sdks/python/setup.py]: see. will do

Comment (markflyhigh) [inline sdks/python/setup.py]: done

**Comment by aaltay** [comment]
LGTM

**Pull request closed without merging**